services/distribution/src/main/java/app/coronawarn/server/services/distribution/runner/TestDataGeneration.java

```diff
@@ -81,8 +78,12 @@
   private static final long TEN_MINUTES_INTERVAL_SECONDS = TimeUnit.MINUTES.toSeconds(10);
 
   @Autowired
-  public TestDataGeneration(DiagnosisKeyService diagnosisKeyService) {
+  public TestDataGeneration(DiagnosisKeyService diagnosisKeyService,
+      DistributionServiceConfig distributionServiceConfig) {
     this.diagnosisKeyService = diagnosisKeyService;
+    this.retentionDays = distributionServiceConfig.getRetentionDays();
```

Comment:
Wasnt sure if this would be an acceptable pattern but started with it thinking this would be least intrusive change to service/controller layer. 

Would love to see/explore better design.

(Applies to most changes)